autograder/cli/users/upsert-file.py

```diff
@@ -0,0 +1,114 @@
+    autograder.cli.common.list_user_op_responses(result['results'], table = arguments['table'])
+    return 0
+
+def _load_users(path):
+    users = []
+
+    with open(path, 'r') as file:
+        lineno = 0
+        for line in file:
+            lineno += 1
+
+            line = line.strip()
+            if (line == ""):
+                continue
+
+            parts = line.split("\t")
+            if (len(parts) > 7):
+                raise ValueError(
+                    "File ('%s') line (%d) has too many values. Max is 7, found %d." % (
+                        path, lineno, len(parts)))
+
+            email = parts.pop(0)
+
+            password = ''
+            if (len(parts) > 0):
+                password = parts.pop(0)
+                if (password != ''):
+                    password = autograder.util.hash.sha256_hex(password)
+
+            name = ''
+            if (len(parts) > 0):
+                name = parts.pop(0)
+
+            role = 'user'
+            if (len(parts) > 0):
+                role = parts.pop(0)
+
+            if (role not in autograder.api.constants.SERVER_ROLES):
+                raise ValueError(
+                    "File ('%s') line (%d) has an invalid role '%s'." % (
+                        path, lineno, role))
+
+            course = ''
+            if (len(parts) > 0):
+                course = parts.pop(0)
+
+            course_role = 'unknown'
+            if (len(parts) > 0):
+                course_role = parts.pop(0)
+
+            if (course_role not in autograder.api.constants.COURSE_ROLES):
+                raise ValueError(
+                    "File ('%s') line (%d) has an invalid course role '%s'." % (
+                        path, lineno, course_role))
+
+            course_lms_id = ''
+            if (len(parts) > 0):
+                course_lms_id = parts.pop(0)
+
+            users.append({
+                'email': email,
+                'pass': password,
+                'name': name,
+                'role': role,
+                'course': course,
+                'course-role': course_role,
+                'course-lms-id': course_lms_id,
+            })
+
+    return users
+
+def main():
+    return run(_get_parser().parse_args())
+
+def _get_parser():
+    parser = autograder.api.users.upsert._get_parser()
+
+    parser.description = ('Upsert users to the course from a TSV file.'
+                + ' (Update if exists, otherwiese insert).')
+
+    parser.add_argument('path', metavar = 'PATH',
+        action = 'store', type = str,
+        help = 'Path to a TSV file where each line contains up to seven columns:'
+                + ' [email, pass, name, role, course, course-role, lms-id].'
+                + ' Only the email is required.')
+
+    parser.add_argument('--skip-emails', dest = 'skip-emails',
+        action = 'store_true', default = False,
+        help = 'Skip sending any emails. Be aware that this may result in inaccessible'
+                + ' information.')
```

Comment:
Default.

Note that we don't need a default on every field,
generally only ones that don't take empty values.

Like in users/upsert, you have most of the "new-" fields without default,
but you include it on the roles because those will always take some value (which is correct).